src/Features/VisualBasic/Portable/AddAccessibilityModifiers/VisualBasicAddAccessibilityModifiersDiagnosticAnalyzer.vb

```diff
@@ -55,13 +55,46 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.AddAccessibilityModifiers
                 Return
             End If
 
-            ' If they already have accessibility, no need to report anything.
+            ' This analyzer bases all of its decisions on the accessibility
             Dim Accessibility = generator.GetAccessibility(member)
-            If Accessibility <> Accessibility.NotApplicable Then
-                Return
+
+            ' Omit will flag any accesibility values that exist and are default
+            ' The other options will remove or ignore accessibility
+            Dim isOmit = [option].Value = AccessibilityModifiersRequired.OmitIfDefault
+
+            If isOmit Then
+                If Accessibility = Accessibility.NotApplicable Then
+                    Return
+                End If
+           
+                If member.IsParentKind(SyntaxKind.CompilationUnit) OrElse
+                   member.IsParentKind(SyntaxKind.NamespaceBlock) Then
+                    ' default is Friend
+                    If Accessibility <> Accessibility.Friend Then
+                        Return
+                    End If
+                End If
+
+                If member.IsParentKind(SyntaxKind.ClassBlock) Then
+                    ' default for const and field in a class is private
+                    If member.IsKind(SyntaxKind.FieldDeclaration) Then
+                        If Accessibility <> Accessibility.Private Then
+                            Return
+                        End If
+                    End If
+                End If
+
+                ' Everything else has a default of public
+                If Accessibility <> Accessibility.Public Then
```

Comment:
the bug is here.  Let's say you have a "Friend class" in a namespace.  First, you do the ```If member.IsParentKind(SyntaxKind.CompilationUnit) OrElse member.IsParentKind(SyntaxKind.NamespaceBlock)``` check.  Since hte member is friend, it's no problem and things fall through.  But then you hit this if-block and decide to not report a diagnostic for this class-node.

Reply:
I would recommend writing is like:

```vb
If isOmit Then
                If Accessibility = Accessibility.NotApplicable Then
                    ' Accessibility modifier already missing.  nothing we need to do.
                    Return
                End If

                If Not MatchesDefaultAccessibility(Accessibility, member) Then
                    ' Explicit accessibility was different than the default accessibility.
                    ' We have to keep this here.
                    Return
                End If
            Else
```

Then implement MatchesDefaultAccessibility simply.

Reply:
The reason this is causing the test to fail is that the currentlogic doesn't report a diagnostic on ```Friend class {|FixAllInDocument:C|}```.  Because there's no diagnostic that is reported that intersects with that span (the ```{|...|}```) the test does nothing.

Reply:
Thanks for the analysis, I'll get this fixed up.